src/sage/graphs/generators/random.py

```diff
@@ -1460,6 +1460,116 @@ def RandomTreePowerlaw(n, gamma=3, tries=1000, seed=None):
         return False
 
 
+def RandomKTree(n, k, seed=None):
+    r"""
+    Returns a random k-tree on `n` nodes numbered `0` through `n-1`.
+
+    ALGORITHM:
+
+    The algorithm first generates a complete graph on `k+1` vertices. Vertices are
+    subsequently generated by randomly choosing one of the existing cliques in the graph,
+    and creating a new clique by replacing one of the vertices in the selected clique with 
+    a newly created one.
+
+    INPUT:
+
+    -  ``n`` -- number of vertices in the k-tree
+
+    -  ``k`` -- within a clique each vertex is connected to `k` vertices. `k` also corresponds
+      to the treewidth of the k-tree
+
+    - ``seed`` -- a ``random.Random`` seed or a Python ``int`` for the random
+      number generator (default: ``None``)
+
+    TESTS::
+        sage: g=graphs.RandomKTree(50,5)
+        sage: g.size()
+        235
+        sage: len(g.vertices(sort=False))
+        50
+        sage: g.treewidth()
+        5
+
+    EXAMPLES::
+
+        sage: G = graphs.RandomKTree(50,5)
+        sage: G.treewidth()
+        5
+        sage: G.show()      
+    """
+
+    from sage.graphs.generators.basic import CompleteGraph
+    
+    if (n<k+1):
+        raise ValueError("n must be greater than k + 1.")
+
+    g = CompleteGraph(k+1)
+    g.name("")
+    
+    cliques = [list(range(k+1))]
+
+    # Randomly choose a row, and copy 1 of the cliques
+    # One of those vertices is then replaced with a new vertex 
+    for i in range(n-k-1):
+        newVertex = k+i+1
+        copiedClique = cliques[randint(0, len(cliques)-1)].copy()
+        copiedClique[randint(0, k)] = newVertex 
+        cliques.append(copiedClique)
+        for u in copiedClique:
+            if u != newVertex:
+                g.add_edge(u, newVertex)
+    return g    
+
+
+def RandomPartialKTree(n, k, x, seed=None):
+    r"""
+    Forms a random k-tree on `n` nodes, and then removes `x` random edges from it leaving
+    a partial k-tree.
+
+    INPUT:
+
+    -  ``n`` -- number of vertices in the k-tree
+
+    -  ``k`` -- within a clique each vertex is connected to `k` vertices. `k` also corresponds
+      to the treewidth of the k-tree
+
+    -  ``x`` -- how many vertices are deleted from the k-tree
+
+    - ``seed`` -- a ``random.Random`` seed or a Python ``int`` for the random
+      number generator (default: ``None``)
+
+    TESTS::
+        sage: g=graphs.RandomPartialKTree(50,5,5)
+        sage: len(g.vertices(sort=False))
+        50
+        sage: g.size()
+        230
+        sage: g.treewidth()
+        5
+
+    EXAMPLES:
+
+        sage: G = graphs.RandomKTree(50,5)
+        sage: G.treewidth()
+        5
+        sage: G.show(
+    """
+    if seed is not None:
+        set_random_seed(seed)
+
+    g = RandomKTree(n,k,seed)
+
+    # Check that x doesn't delete too many edges
+    if x > g.size():
+        raise ValueError("x must be less than the number of nodes in the K-Tree with `n` nodes.")
```

Comment:
number of nodes or edges ? you compared `x` to the number of edges of `g`.  Also, avoid the final `.`

Reply:
Yep, should be edges